core/vm/contracts_state.go

```diff
@@ -10,45 +10,46 @@ import (
 	stakingTypes "github.com/harmony-one/harmony/staking/types"
 )
 
-// WriteCapablePrecompiledContractsStaking lists out the write capable precompiled contracts
+// StateDependentPrecompiledContractsStaking lists out the write capable precompiled contracts
 // which are available after the StakingPrecompileEpoch
-// for now, we have only one contract at 252 or 0xfc - which is the staking precompile
-var WriteCapablePrecompiledContractsStaking = map[common.Address]WriteCapablePrecompiledContract{
+var StateDependentPrecompiledContractsStaking = map[common.Address]StateDependentPrecompiledContract{
 	common.BytesToAddress([]byte{252}): &stakingPrecompile{},
 }
 
-// WriteCapablePrecompiledContract represents the interface for Native Go contracts
+// StateDependentPrecompiledContractsStaking lists out the write capable precompiled contracts
+// which are available after the ROStakingPrecompileEpoch
+var StateDependentPrecompiledContractsRoStaking = map[common.Address]StateDependentPrecompiledContract{
+	common.BytesToAddress([]byte{252}): &stakingPrecompile{},
+	common.BytesToAddress([]byte{250}): &roStakingPrecompile{},
```

Comment:
Why is it not added to the readOnly contracts?

Reply:
To be able to inject EVM / state object. Instead of calling it `contracts_write` this change renames them to `contracts_state` - which need state to either read or write. The write capability is, of course, checked in each precompile.